build-aux/snap/snapcraft.yaml

```diff
@@ -36,23 +37,93 @@ parts:
     # these packages are needed to call mkversion.sh in override-pull, all other
     # dependencies are installed using apt-get build-dep
     build-packages:
-      - git
       - dpkg-dev
+      - git
+      - sudo
+      - wget
     after: [apparmor]
     override-pull: |
       snapcraftctl pull
-      # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
-      snapcraftctl set-version "$(./mkversion.sh --output-only)"
       # Ensure that ./debian/ packaging which we are about to use
       # matches the current `build-base` release. I.e. ubuntu-16.04
       # for build-base:core, etc.
       ./generate-packaging-dir
       # install build dependencies
       export DEBIAN_FRONTEND=noninteractive
       export DEBCONF_NONINTERACTIVE_SEEN=true
-      sudo -E apt-get build-dep -y ./
+      sudo -E apt-get build-dep -y "$SNAPCRAFT_PART_SRC"
+      # Get libc6 from xenial. Note that we are not shipping these, we
+      # just need to build against them so that only symbols for the
+      # xenial version of glibc are used by go. Later in the "libc6"
+      # part of snapcraft.yaml the most recent modern libc6 is put
+      # into the snapd.snap. The build against old symbols is only
+      # needed to ensure that the snapd snap can start on 16.04 Ubuntu
+      # system. As glibc is full backwards compatible in newer
+      # versions this ensure that it also works on any newer
+      # distro/libc6 combination.
```

Comment:
This is still sort of a stop-gap that cannot be maintained forever. the libc6 here is still not the one ESM patched, and building against point in time glibc, can still generate calls to the versions of APIs that have become vulnerable and would be fixed by building against a glibc with updated headers or optional symbols that would attempt to do secure calls by default (if/when available at runtime).

Also even this version of libc6 will not be available from that location for much longer.